### PR TITLE
56/create wallet api

### LINKED ISSUE #56: Create Wallet API and mock implementation
Create Wallet API and mock implementation

More info in the PR #57

### PR DESCRIPTION
**Pull request by anxolin**
Defines the WalletAPI, that abstracts the access to an ethereum wallet.

Fixes #56 

**Context**
The idea is to: 
1. Create an interface in Typescript defining the method signatures
2. Provide a simple, "in memory", pure JS implementation. This is for making development simpler. We will be able to start the app using this mock implementation.

**Main components**
* `WalletApi`: The typescript interface. Defines the basic methods for accessing an ethereum wallet, specially the ones we need for this spring (`getAddress()` and `getNetworkId()`)
* `WalletApiMock`: Basic implementation of this API

**Aditional**
* Although not strictly related to the PR, allow me to include a fix on a prior PR. It allows to define a `MOCK` env var that controls whether we run the app in mock mode, or in REAL mode.

**Not included**
* There's a TODO that fakes the connection of the wallet. The idea is to remove it once we implement the "connect wallet" functionality (look for the TODO of `await walletApi.connect()`)

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr57--dexreact.review.gnosisdev.com

File: src/api/WalletApi/WalletApiMock.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { WalletApi, Network } from 'types'
+import BN from 'bn.js'
+
+import { ADDRESS } from '../../../test/data'
+
+/**
+ * Basic implemtation of Wallet API
```
Comment (alfetopito) [inline src/api/WalletApi/WalletApiMock.ts]: typo: `implemtation`

File: src/api/index.ts [hunk 2]
```diff
@@ -1,22 +1,32 @@
...
 
 function createTokenApi(): TokenApi {
   return new TokenApiImpl([Network.Mainnet, Network.Rinkeby])
 }
 
+function createWalletApi(): WalletApi {
+  if (isMock) {
+    return new WalletApiMock()
+  } else {
+    throw new Error('Only has been implemented')
```
Comment (alfetopito) [inline src/api/index.ts]: Only what? :)

File: src/api/index.ts [hunk 3]
```diff
@@ -1,22 +1,32 @@
...
+  }
+}
+
 function createDepositApi(): DepositApi {
-  if (true || isMock) {
+  if (isMock) {
     return new DepositApiMock({ balanceStates })
   } else {
     throw new Error('Only has been implemented')
```
Comment (alfetopito) [inline src/api/index.ts]: Same as previous comment.

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr57--dexreact.review.gnosisdev.com

File: src/api/index.ts (hunk 2 above)
Comment (anxolin) [reply src/api/index.ts]: Yes, this was corrected after you mentioned it in a PR. It's solved in https://github.com/gnosis/dex-react/pull/57/commits/f672573b33711bdcffacfd0ee204c8f58c9ae451

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr57--dexreact.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr57--dexreact.review.gnosisdev.com

**Pull request merged**